### PR TITLE
Fix unused (and so, broken) enable_rollback_recording.

### PR DESCRIPTION
**Pull request by nerzhul**
enable_rollback_recording is always enabled whereas option is disabling it by default.
This commit prevent the creation of RollbackManager into server.cpp which is the test done by all server components to see if rollbackmanager must be used (a better settings interface is needed to replace this ugly test...).

This option isn't reloaded at each server loop but loaded when server starts, for data consistency

### CODE REVIEW

File: src/script/lua_api/l_rollback.cpp [hunk 1]
```diff
@@ -80,6 +83,14 @@ int ModApiRollback::l_rollback_revert_actions_by(lua_State *L)
 	int seconds = luaL_checknumber(L, 2);
 	Server *server = getServer(L);
 	IRollbackManager *rollback = server->getRollbackManager();
+
+	// If rollback is disabled, tell it's not a success.
+	if (rollback == NULL) {
+		bool success = false;
+		lua_pushboolean(L, success);
+		lua_createtable(L, 0, 0);
```
Comment (ShadowNinja) [inline src/script/lua_api/l_rollback.cpp]: `lua_newtable(L)` is a smightly simpler alias for this.

File: src/script/lua_api/l_rollback.cpp [hunk 2]
```diff
@@ -80,6 +83,14 @@ int ModApiRollback::l_rollback_revert_actions_by(lua_State *L)
 	int seconds = luaL_checknumber(L, 2);
 	Server *server = getServer(L);
 	IRollbackManager *rollback = server->getRollbackManager();
+
+	// If rollback is disabled, tell it's not a success.
+	if (rollback == NULL) {
+		bool success = false;
+		lua_pushboolean(L, success);
```
Comment (ShadowNinja) [inline src/script/lua_api/l_rollback.cpp]: The success variable is pointless, just do `lua_pushboolean(L, false);  // Success` or so.

File: src/server.cpp [hunk 3]
```diff
@@ -406,7 +409,8 @@ Server::~Server()
 	// N.B. the EmergeManager should be deleted after the Environment since Map
 	// depends on EmergeManager to write its current params to the map meta
 	delete m_emerge;
-	delete m_rollback;
+	if (m_rollback)
+		delete m_rollback;
```
Comment (ShadowNinja) [inline src/server.cpp]: This check is unnecessary, `delete` is a no-op on `NULL` pointers.

File: builtin/game/chatcommands.lua [hunk 4]
```diff
@@ -583,6 +583,10 @@ core.register_chatcommand("rollback_check", {
 			local name = puncher:get_player_name()
 			core.chat_send_player(name, "Checking " .. core.pos_to_string(pos) .. "...")
 			local actions = core.rollback_get_node_actions(pos, range, seconds, limit)
+			if actions == nil then
```
Comment (ShadowNinja) [inline builtin/game/chatcommands.lua]: 1. This should be `if not actions then`.
  2. This should quick-fail as soon as the command is run if rollback recording is disabled.

**Comment by ShadowNinja** [comment]
Has a few nits, but pretty good.

:+1:

**Comment by nerzhul** [comment]
Then maybe adding a lua call to check if rollback is enable ?

**Comment by ShadowNinja** [comment]
@nerzhul Just using `setting_get` should be good enough.

**Comment by nerzhul** [comment]
simple and efficient :) i'll look at this after merging some blocker PR. I'll improve this PR too and we need to merge it in master. Do you want a merge into feature freeze too ? you added a milestone

**Comment by ShadowNinja** [comment]
@nerzhul Yes, I think this is a pretty important bugfix.

**Comment by nerzhul** [comment]
Okay, i improve it for merge

**Comment by nerzhul** [comment]
Ok @ShadowNinja http://irc.minetest.ru/minetest-dev/2015-02-18#i_4159055

**Comment by nerzhul** [comment]
commits 4208fdfd2273b715eeddb8b9877c2def8346d447 and 27d4e89d326337c8d8cafc681acde8bfbde7de77

**Pull request closed without merging**